app/src/Controller/LdapController.php

```diff
@@ -0,0 +1,226 @@
+     *
+     * @SuppressWarnings(PHPMD.StaticAccess)
+     *
+     * @return JsonResponse
+     */
+    public function getLdapEntries(
+        Client $ldap,
+        Request $request,
+        SerializerInterface $serializer
+    ): JsonResponse {
+
+        $query = $request->get('query', '(objectClass=*)');
+        $baseDn = $request->get('base', null);
+
+        $options = [];
+
+        $filters = json_decode($request->get('filters', '[]'), true);
+        if (!empty($filters)) {
+            $queryFilters = '';
+            foreach ($filters as $field => $value) {
+                switch ($field) {
+                    case 'dn':
+                        $queryFilters .= "($value)";
+                        break;
+
+                    default:
+                        $queryFilters .= "($field=$value)";
+                        break;
+                }
+            }
+            $query = "(&$queryFilters$query)";
+        }
+        // XXX Is there any way to ask LDAP server to sort results?
+        //$orders  = json_decode($request->get('orders', null), true);
+
+        // Attributes option.
+        $attributes = $request->get('attributes', ['dn']);
+        $options['filter'] = $attributes;
+
+        // Add query options.
+        $size = (int) $request->get('size', 0);
+        $max = (int) $request->get('max', 0);
+
+        $options['pageSize'] = $size;
+        $options['maxItems'] = $max;
+
+        try {
+            $ldap->bind();
+            $ldapEntries = $ldap->search($query, $baseDn, $options);
+        } catch (LdapException $exception) {
+            return new JsonResponse($exception->getMessage(), 500);
+        }
+        
+        $total = count($ldapEntries);
+        $entries = array();
+
+        foreach ($ldapEntries as $key => $ldapEntry) {
+            $dto = LdapEntryDTO::fromEntry($ldapEntry);
+
+            $entries[$key]['dn'] = $dto->getDn();
+            $entries[$key]['attributes'] = $dto->getAttributes();
+        }
+
+        $entries = $serializer->normalize(
+            $entries,
+            Entry::class
+        );
+
+        return new JsonResponse([
+            'total' => $total,
+            'items' => $entries,
+        ]);
+    }
+
+    /**
+     * @Route("/api/ldap/{fullDN}", name="get_ldap_entry", methods={"GET"})
+     *
+     * @SuppressWarnings(PHPMD.StaticAccess)
+     *
+     * @return JsonResponse Entry | LdapException
+     */
+    public function getLdapEntryByDn(
+        string $fullDN,
+        Client $ldap,
+        Request $request,
+        SerializerInterface $serializer
+    ): JsonResponse {
+
+        $query = $request->get('query', '(objectClass=*)');
+
+        $options = $request->get('options', []);
+
+        try {
+            $ldap->bind();
+            $ldapEntry = $ldap->get($query, $fullDN, $options);
+        } catch (LdapException $exception) {
+            return new JsonResponse($exception->getMessage(), 400);
```

Comment:
The `getLdapEntries` function throws a 500 in this situation but here we throw a 400... Not sure if it's a good thing consistency wise.